server/src/main/java/org/elasticsearch/index/query/BoolQueryBuilder.java

```diff
@@ -384,30 +384,14 @@ protected Query doToQuery(QueryShardContext context) throws IOException {
             return new MatchAllDocsQuery();
         }
 
-        final String minimumShouldMatch;
-        if (context.isFilter() && this.minimumShouldMatch == null && shouldClauses.size() > 0) {
-            minimumShouldMatch = "1";
```

Comment:
This is the part that confuses me. Why do we need to set `minimum_should_match` in this case ? I cannot think of one example where this fixes something ? @martijnvg can you confirm ?